TELCODOCS-1576: Image-based upgrade with Lifecycle Agent (#69809)
openshift/openshift-docs: modules/ztp-image-based-upgrade-rollback.adoc

// Module included in the following assemblies:
// Epic TELCOSTRAT-160 (4.15/4.16), story TELCODOCS-1576
// * scalability_and_performance/ztp-talm-updating-managed-policies.adoc

:_mod-docs-content-type: PROCEDURE
[id="ztp-image-based-upgrade-rollback_{context}"]
= (Optional) Initiating rollback of the {sno} cluster after an image-based upgrade

By default, an automatic rollback is enabled in case the upgrade fails at any point.
You can disable the automatic rollback in the `ImageBasedUpgrade` CR at the `Prep` stage:

> +

Reviewer: the `+` is getting rendered in the output doc

// Module included in the following assemblies:
// Epic TELCOSTRAT-160 (4.15/4.16), story TELCODOCS-1576
// * scalability_and_performance/ztp-talm-updating-managed-policies.adoc

:_mod-docs-content-type: PROCEDURE
[id="ztp-image-based-upgrade-rollback_{context}"]
= (Optional) Initiating rollback of the {sno} cluster after an image-based upgrade

By default, an automatic rollback is enabled in case the upgrade fails at any point.
You can disable the automatic rollback in the `ImageBasedUpgrade` CR at the `Prep` stage:

.Example ImageBasedUpgrade CR
[source,yaml]
----
apiVersion: lca.openshift.io/v1alpha1
kind: ImageBasedUpgrade
metadata:
  name: example-upgrade
spec:
  stage: Idle
  seedImageRef:
    version: 4.14.6
    image: <seed_container_image>
  additionalImages:
    name: ""
    namespace: ""
  autoRollbackOnFailure: {} <1>
#    disabledForPostRebootConfig: "true" <2>
#    disabledForUpgradeCompletion: "true" <3>
#    disabledInitMonitor: "true" <4>
#    initMonitorTimeoutSeconds: 1800 <5>
[...]
----

> <1> By default, automatic rollback is enabled throughout the upgrade.

Reviewer 2: ```suggestion
<1> By default, automatic rollback on failure is enabled throughout the upgrade.
```

// Module included in the following assemblies:
// Epic TELCOSTRAT-160 (4.15/4.16), story TELCODOCS-1576
// * scalability_and_performance/ztp-talm-updating-managed-policies.adoc

:_mod-docs-content-type: PROCEDURE
[id="ztp-image-based-upgrade-rollback_{context}"]
= (Optional) Initiating rollback of the {sno} clusters after an image-based upgrade

If you encounter unresolvable issues after an upgrade, you can initiate a rollback by using {cgu-operator}.

.Prerequisites

* You have installed the OpenShift CLI (`oc`).

* You have logged in to the hub cluster as a user with `cluster-admin` privileges.

* You have installed {cgu-operator}.

> 

Reviewer 3: * You are halfway through an upgrade on a managed cluster.

Reviewer 3: or similar...

// are there other prereqs?

.Procedure

. Move to the next stage by changing the `Upgrade` value to `Rollback` in the `ImageBasedUpgrade` CR.
+
[source,yaml]
----
apiVersion: lca.openshift.io/v1alpha1
kind: ImageBasedUpgrade
metadata:
  name: example-upgrade
spec:
  stage: Rollback
  seedImageRef:
    version: 4.14.0

>     image: quay.io/<seed_image_repository>

Reviewer 3: ```suggestion
  image: <seed_container_image>
```

----
// For telco, we need a sample with extraManifests and oadpContent

. Apply the `ImageBasedUpgrade` policy with {cgu-operator}.
+
[source,terminal]
----
$ oc apply -f example-upgrade.yml
----
// Please check command

> The {lcao} reboots the node with the previous release of {product-title} and applications.

Reviewer 3: ```suggestion
The {lcao} reboots the node with the previously installed version of {product-title}.
```

re: applications - what about workloads? Are workloads also reapplied?

. Commit to the rollback by changing the `Rollback` value to `Idle` in the `ImageBasedUpgrade` CR.
+
[source,yaml]
----
apiVersion: lca.openshift.io/v1alpha1
kind: ImageBasedUpgrade
metadata:
  name: example-upgrade
spec:
  stage: Idle
  seedImageRef:
    version: 4.14.0

>     image: quay.io/<seed_image_repository>

Reviewer 3: ```suggestion
    image: <seed_container_image>
```
